pkg/costmodel/cluster.go

```diff
@@ -702,7 +703,8 @@ func ClusterLoadBalancers(cp cloud.Provider, client prometheus.Client, duration,
 // ComputeClusterCosts gives the cumulative and monthly-rate cluster costs over a window of time for all clusters.
 func (a *Accesses) ComputeClusterCosts(client prometheus.Client, provider cloud.Provider, window, offset string, withBreakdown bool) (map[string]*ClusterCosts, error) {
```

Comment:
Looping in @nikovacevic here because I know he normalize a lot of the window/offset input, but should we aim to accept a `time.Duration` in the non-http handler functions? With the motivation to remove the awkward inclusion of a `timeutil.CleanDurationString` call? The http handlers can use `queryParams.GetDuration(...)` (which should include CleanDurationString internally), and the rest of the APIs can just reliably accept a `time.Duration`. Appreciate thoughts from @michaelmdresser and @Sean-Holcomb as well.

Reply:
I'm onboard with that idea. I already felt like I was getting out of scope for this bug fix with the package change, but that sounds like the right approach to something I felt was a little awkward too. Just sounds like some more hack and slash clean up.

Reply:
Agree that `timeutil.CleanDurationString` is awkward, and requires every engineer both to know and to remember to call that function all the time. Ideally this kind of thing would be patched in one place and "just work." (For instance, in a previous review of this PR, I asked why this logic couldn't simply be added to the beginning of `ParseTimeRange` for only the `offset string` parameter, and never got a response, but the question stands.)

I'm 100% in favor of @mbolt35's recommendation of only allowing `window` and `offset` to be passed as strings where required by the circumstance (e.g. HTTP parameters) and to be of type `time.Duration` everywhere else. That said, I also don't want to blow up scope.

@Sean-Holcomb in your judgment, is this a pervasive and serious enough issue to warrant a deep clean up effort right now? Or can we fix the originally-scoped bug with a cleaner, simpler implementation and leave deep-cleaning refactors of duration and offset strings for a future clean-up task?

Reply:
I think it's worth the time if you're already in there - This already makes a big difference IMO - the duplication of this method in a few places was starting to become unwieldy.

Reply:
@nikovacevic sorry for not responding to that comment, but you should know that it did lead me to delete some more duplicate code. Moving it ParseTimeRange method doesn't really change the need for the awkward placements at higher up in the call chain, because the string itself is causing problems when it is used for Prometheus queries it becomes "offset offset 3h". Both of this methods return a time that is used in a different way

Reply:
Also strongly in favor of parsing strings into useful types in advance so we don't have to later. I'll defer to @Sean-Holcomb to determine the scoping of immediately changing this across the codebase vs. delaying.